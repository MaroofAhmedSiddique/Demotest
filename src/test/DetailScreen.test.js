import React from 'react';
import renderer from 'react-test-renderer';
import {TouchableOpacity} from 'react-native';
import PostDetailsScreen from '../screens/PostdetailScreen'; // Import your component

// Mock navigation props
const navigation = {
  goBack: jest.fn(),
};

describe('PostDetailsScreen', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <PostDetailsScreen
          route={{params: {item: {}}}}
          navigation={navigation}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls goBack when back button is pressed', () => {
    const component = renderer.create(
      <PostDetailsScreen
        route={{params: {item: {}}}}
        navigation={navigation}
      />,
    );
    const backButton = component.root.findByType(TouchableOpacity);
    backButton.props.onPress();
    expect(navigation.goBack).toHaveBeenCalled();
  });
});
