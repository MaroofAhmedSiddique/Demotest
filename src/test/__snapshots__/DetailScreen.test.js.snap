// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostDetailsScreen renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "backgroundColor": "#f5f5f5",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../src/assets/ic_backIcon.png",
        }
      }
      style={{}}
    />
  </View>
  <RCTScrollView
    bounces={false}
  >
    <View>
      <View
        style={
          {
            "flexDirection": "row",
            "gap": 10,
            "marginTop": 20,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 20,
            }
          }
        >
          Item Page 
        </Text>
      </View>
      <View
        style={
          {
            "marginTop": 25,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 25,
              "fontWeight": "bold",
            }
          }
        >
          Title
        </Text>
        <Text
          style={
            {
              "color": "#555",
              "fontSize": 17,
              "lineHeight": 24,
            }
          }
        />
      </View>
      <View
        style={
          {
            "marginTop": 25,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 25,
              "fontWeight": "bold",
            }
          }
        >
          Body
        </Text>
        <Text
          style={
            {
              "color": "#555",
              "fontSize": 17,
              "lineHeight": 24,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
